lms/djangoapps/instructor/views/api.py

```diff
@@ -2707,10 +2707,141 @@ def start_certificate_regeneration(request, course_id):
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @require_global_staff
+@require_http_methods(['POST', 'DELETE'])
+def certificate_exception_view(request, course_id):
+    """
+    Add/Remove students to/from certificate white list.
+
+    :param request: HttpRequest object
+    :param course_id: course identifier of the course for whom to add/remove certificates exception.
+           deleting certificate exception, it will be empty if new certificate exception is being created.
+    :return: JsonResponse object with success/error message or certificate exception to sync on the frontend.
+    """
+    course_key = CourseKey.from_string(course_id)
+    # Validate request data and return error response in case of invalid data
+    try:
+        certificate_exception, student = parse_request_data(request)
+    except ValueError as error:
+        return JsonResponse({'success': False, 'message': error.message}, status=400)
+
+    # Add new Certificate Exception for the student passed in request data
+    if request.method == 'POST':
+        try:
+            exception = add_certificate_exception(course_key, student, certificate_exception)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+        return JsonResponse(exception)
+
+    # Remove Certificate Exception for the student passed in request data
+    elif request.method == 'DELETE':
+        try:
+            remove_certificate_exception(course_key, student)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+
+        return JsonResponse({})
+
+
+def add_certificate_exception(course_key, student, certificate_exception):
+    """
+    Add a certificate exception to CertificateWhitelist table. Raises ValueError in case of invalid data.
+
+    :param course_key: course key of the course for whom to add certificates exception
+    :param student: User object for whom to add Certificate Exception
+    :param certificate_exception: JSON representation of backbone CertificateException model
+    :return: CertificateWhitelist item in dict format containing all the keys required by
+             backbone model CertificateException.
+    """
+    if CertificateWhitelist.objects.filter(user=student, course_id=course_key, whitelist=True).count() > 0:
```

Comment:
@saleem-latif you implemented a method `get_certificate_white_list` in certificate model, it is returning a list of white listed users. I think we can update this method for a user as well e.g. `def get_certificate_white_list(cls, course_id, student=none):` 
we can use this method to get all white list users against the given course OR a record for a given user.